### PR TITLE
Improvement/zenko 4525 lint other tests

### PR DESCRIPTION
**Pull request by KazToozs**
~~Discussion with @williamlardier asks whether we should remove the `@typescript-eslint/no-unsafe-assignment` rule from the linter~~

### CODE REVIEW

File: tests/ctst/steps/notifications.ts [hunk 1]
```diff
@@ -49,41 +76,51 @@ async function emptyVersionedBucket(world: IWorld) {
     }));
 }
 
-async function putObject(world: IWorld) {
+async function putObject(world: Zenko) {
```
Comment (ghost) [inline tests/ctst/steps/notifications.ts:84]: Mixed with [the logic we currently have](https://github.com/scality/Zenko/pull/1820/files#diff-504fd6b6f773cd1c920c1a4789386463c448655cdeb9ca7395ae4108cd787d6aR8), the World may not always be Zenko...

File: tests/ctst/steps/notifications.ts [hunk 2]
```diff
@@ -49,41 +76,51 @@ async function emptyVersionedBucket(world: IWorld) {
...
     world.addCommandParameter({ key: objName });
-    world.addCommandParameter({ copySource: `${world.saved.bucketName}/${world.saved.objectName}` });
-    world.saved.objectName = objName;
+    world.addCommandParameter({
+        copySource:
+            `${world.getSaved().bucketName as string}/${world.getSaved().objectName as string}`,
+    });
+    // TODO this rule honestly just seems dumb, especially in this case
+    // eslint-disable-next-line no-param-reassign
+    world.getSaved().objectName = objName;
```
Comment (ghost) [inline tests/ctst/steps/notifications.ts]: We should have a `setSaved()` method in this case, it does not feel standard to edit the returned variable of a `get` function in the same line

The approach before looked more standard

File: tests/ctst/package.json
Comment (ghost) [inline tests/ctst/package.json:1]: About the linter rule: if it works well today, it's ok to keep it. With the latest changes you enforces a code style that averyone must follow, so this should be all good. Let's wait and see if the rule is too restrictive

File: tests/ctst/world/Zenko.ts [hunk 3]
```diff
@@ -169,8 +170,8 @@ export default class Zenko extends World<ZenkoWorldParameters> {
      * @param {object} result - contains both stderr and stdout from the CLI command.
      * @returns {boolean} - if the result is a success or a failure
      */
-    checkResult(result: Utils.Command | Utils.Command[]): boolean {
-        const usedResult: Utils.Command[] = Array.isArray(result) ? result : [result];
+    checkResult(result: ApiResult | ApiResult[]): boolean {
+        const usedResult: ApiResult[] = Array.isArray(result) ? result : [result];
```
Comment (ghost) [inline tests/ctst/world/Zenko.ts]: Alternatively we could rename it `checkResults` and only accept arrays, so we can only call it with something like `checkResult([myResultObject])` if we only have one

File: tests/ctst/steps/notifications.ts [hunk 4]
```diff
@@ -92,168 +131,182 @@ async function putTag(world: IWorld) {
...
-    this.saved.notificationDestinations.push({
-        destinationName: this.parameters.NotificationDestination,
-        topic: this.parameters.NotificationDestinationTopic,
-        hosts: this.parameters.KafkaHosts,
+Given('two notification destinations', function (this: Zenko) {
+    const notificationDestinations = [];
+    notificationDestinations.push({
+        destinationName: (this.parameters as { NotificationDestination: string }).NotificationDestination,
+        topic: (this.parameters as { NotificationDestinationTopic: string }).NotificationDestinationTopic,
+        hosts: (this.parameters as { KafkaHosts: string }).KafkaHosts,
```
Comment (ghost) [inline tests/ctst/steps/notifications.ts]: Can we better type these variables using what is defined in the Zenko world?

**Review by ghost — APPROVED** [review]

**Review by XinLi0207 — APPROVED** [review]

**Comment by KazToozs** [comment]
ping

**Pull request merged**